src/work-allocation-2/components/case-manager-filter/case-manager-filter.component.ts

```diff
@@ -1,97 +1,151 @@
-import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
-
-import { SessionStorageService } from '../../../app/services';
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { FilterService } from '@hmcts/rpx-xui-common-lib';
+import { FilterConfig, FilterFieldConfig, FilterSetting } from '@hmcts/rpx-xui-common-lib/lib/models/filter.model';
+import { filter } from 'rxjs/operators';
 import { Location } from '../../models/dtos';
-import { FilterConstants } from '../constants';
 
 @Component({
   selector: 'exui-case-manager-filter',
   templateUrl: './case-manager-filter.component.html',
-  styleUrls: ['./case-manager-filter.component.scss']
+  styleUrls: ['./case-manager-filter.component.scss'],
+  encapsulation: ViewEncapsulation.None,
 })
-export class CaseManagerFilterComponent implements OnChanges {
-  // Protected instances of the exported constants.
-  public readonly ALL_LOCATIONS: Location = FilterConstants.Options.Locations.ALL;
+export class CaseManagerFilterComponent implements OnInit {
 
-  /**
-   * The locations that are available to be selected from.
-   */
-  @Input() public locations: Location[];
+  private static FILTER_NAME: string = 'cases';
 
-  @Input()
-  public get location(): Location {
-    return this.pLocation;
-  }
+  @Input() public locations: Location[] = [];
 
-  public set location(value: Location) {
-    value = value || this.ALL_LOCATIONS; // undefined or null means "All"
-    if (this.pLocation !== value) {
-      this.pLocation = value;
-      this.saveFilterToSession();
-      this.emitChangedEvent();
+  public fieldsConfig: FilterConfig = {
+    persistence: 'session',
+    enableDisabledButton: true,
+    id: CaseManagerFilterComponent.FILTER_NAME,
+    fields: [],
+    cancelButtonText: 'Reset to default',
+    applyButtonText: 'Apply',
```

Comment:
Same texts on task-list-filter.component.ts, could these be extracted to common const keys and reuse them in multiple place.

Reply:
task-list-filter.component.ts has the same text as below:
    
    cancelButtonText: 'Reset to default',
    applyButtonText: 'Apply',

Could these two be extracted to common constants?

Reply:
@kuda1992 this is the last comment